src/test/java/com/exadel/discountwebapp/category/CategoryServiceIntegrationTest.java

```diff
@@ -0,0 +1,114 @@
+package com.exadel.discountwebapp.category;
+
+import com.exadel.discountwebapp.category.entity.Category;
+import com.exadel.discountwebapp.category.repository.CategoryRepository;
+import com.exadel.discountwebapp.category.service.CategoryService;
+import com.exadel.discountwebapp.category.vo.CategoryRequestVO;
+import com.exadel.discountwebapp.category.vo.CategoryResponseVO;
+import com.exadel.discountwebapp.exception.EntityNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.jdbc.Sql;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+@Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:sql/category-init.sql")
+@Sql(executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, scripts = "classpath:sql/clean-up.sql")
+class CategoryServiceIntegrationTest {
+
+    @Autowired
+    private CategoryService categoryService;
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @Test
+    void shouldGetCategoryById() {
+        var id = 1L;
+        var expected = categoryRepository.findById(id).get();
+        var actual = categoryService.findById(id);
+
+        matchOne(expected, actual);
+    }
+
+    @Test
+    void shouldThrowExceptionIfNoCategoryFoundById() {
+        var id = 3L;
+        assertThrows(EntityNotFoundException.class, () -> {
+            categoryService.findById(id);
+        });
+    }
+
+    @Test
+    void shouldGetAllCategories() {
+        var actual = (List<Category>) categoryRepository.findAll();
```

Comment:
It seems not safe to cast Iterable to List. Because it is an internal decision of Spring Data what implementation to use. Can we avoid casting to List here?

Reply:
Created a method for casting